oggm/core/preprocessing/climate.py

```diff
@@ -136,6 +136,230 @@ def process_custom_climate_data(gdir):
+    ts_pre_avg= ts_pre_avg.groupby('time.month').mean(dim='time')
+    ts_pre = precp.groupby('time.month')-ts_pre_avg
+
+    # interpolate to HR grid
+    if np.any(~np.isfinite(ts_tmp[:, 1, 1])):
+        # Extreme case, middle pix is not valid
+        # take any valid pix from the 3*3 (and hope there's one)
+        found_it = False
+        for idi in range(2):
+            for idj in range(2):
+                if np.all(np.isfinite(ts_tmp[:, idj, idi])):
+                    ts_tmp[:, 1, 1] = ts_tmp[:, idj, idi]
+                    ts_pre[:, 1, 1] = ts_pre[:, idj, idi]
+                    found_it = True
+        if not found_it:
+            msg = '{}: OMG there is no climate data'.format(gdir.rgi_id)
+            raise RuntimeError(msg)
+    elif np.any(~np.isfinite(ts_tmp)):
+        # maybe the side is nan, but we can do nearest
+        ts_tmp = ncclim.grid.map_gridded_data(ts_tmp.values, temp.salem.grid,
+                                              interp='nearest')
+        ts_pre = ncclim.grid.map_gridded_data(ts_pre.values, precp.salem.grid,
+                                              interp='nearest')
+    else:
+        # We can do bilinear
+        ts_tmp = ncclim.grid.map_gridded_data(ts_tmp.values, temp.salem.grid, interp='linear')
+
+        ts_pre = ncclim.grid.map_gridded_data(ts_pre.values, precp.salem.grid,
+                                              interp='nearest')
+
+    # take the center pixel and add it to the CRU CL clim
+    # for temp
+    loc_tmp = xr.DataArray(loc_tmp[:, 1, 1], dims=['month'],
+                           coords={'month':ts_tmp_avg.month})
+    ts_tmp = xr.DataArray(ts_tmp[:, 1, 1], dims=['time'],
+                           coords={'time':time})
+    ts_tmp = ts_tmp.groupby('time.month') + loc_tmp
+    # for prcp
+    loc_pre = xr.DataArray(loc_pre[:, 1, 1], dims=['month'],
+                           coords={'month':ts_pre_avg.month})
+    ts_pre = xr.DataArray(ts_pre[:, 1, 1], dims=['time'],
+                           coords={'time':time})
+    ts_pre = ts_pre.groupby('time.month') + loc_pre
+
+    # load dates in right format to save
+    Tindex=salem.GeoNetcdf(fpathTemp, monthbegin=True)
+    time1 = Tindex.variables['time']
+    time2 = time1[9:-3] - Ndays # from normal years to hydrological years
+    time2 = netCDF4.num2date(time2, time1.units, calendar='noleap')
+
+    # done
+    loc_hgt = loc_hgt[1, 1]
+    loc_lon = loc_lon[1]
+    loc_lat = loc_lat[1]
+    assert np.isfinite(loc_hgt)
+    assert np.all(np.isfinite(ts_pre.values))
+    assert np.all(np.isfinite(ts_tmp.values))
+    assert np.all(np.isfinite(ts_grad))
+    gdir.write_gcm_data_file(time2, ts_pre.values, ts_tmp.values,
+                                    ts_grad, loc_hgt, loc_lon, loc_lat)
+
+    ncclim._nc.close()
+    Tindex._nc.close()
+
+    # metadata
+    out = {'climate_source': 'GCM data', 'hydro_yr_0': y0 + 1, 'hydro_yr_1': y1}
+    gdir.write_pickle(out, 'climate_info')
```

Comment:
This overwrites the existing metadata, which isn't desirable. You should remove these two lines for now.